interp/impargs.ml

```diff
@@ -22,6 +22,8 @@ open EConstr
 open Reductionops
 open Constrexpr
 
+let whd_all = Reductionops.clos_whd_flags CClosure.all
```

Comment:
I think it's a fairly bad idea to shadow a global function with a local one that has a subtly different behaviour. It'd be better to use a different name to insist on the changes.

Reply:
OK, but I won't touch this again before bench terminates. I'll call it whd_all_call_by_need.

IMO the right fix would be to deprecated all call by name APIs in reductionops. Why would one call these?

Reply:
> IMO the right fix would be to deprecated all call by name APIs

Nope, non-kernel reduction makes sense sometimes. Sometimes the overhead introduced by the change of representation is so high that the use of call-by-need is irrelevant.

Reply:
Then they should be called `whd_*_by_name`. I pretty sure nobody wanted to call a by name procedure here, imo the names are misleading. If there is a use case for these API, fine, but how many of their users are consciously choosing by name?